client/src/components/list/List.tsx

```diff
@@ -0,0 +1,113 @@
+    }, []);;
+
+    const [colDefs, _ ] = useState<ColDef<KxDocument>[]>([
+        { headerName: "ID", minWidth: 100, enableRowGroup: true, valueFormatter: () => { return ""} },
+        { headerName: "Title", field: "title", minWidth: 170, enableRowGroup: false,  filter: true, sortable: true },
+        { headerName: "Stakeholders", field: "stakeholders", enableRowGroup: false,  filter: true,},
+        { headerName: "Scale", field: "scale", enableRowGroup: true,  filter: true, valueFormatter: (params: { value: string | number; }) => {
+             return params.value !== undefined ? "1:" + params.value.toLocaleString() : ""
+        }},
+        { headerName: "Issuance Date", field: "issuance_date", valueFormatter: (params: { value: string | number; }) => {
+             return params.value !== undefined ? new Date(params.value).toLocaleDateString() : ""
+        }},
+        { headerName: "Type", field: "type", enableRowGroup: true,  filter: true},
+        { headerName: "Language", field: "language", enableRowGroup: true,  filter: true, valueFormatter: (params: { value: string | number; }) => {
+             return locales.find((l) => l.code === params.value)?.name || "" 
+        }},
+        { headerName: "Pages", field: "pages", enableRowGroup: false,  filter: true  },
+    ]);
+    const autoGroupColumnDef = {
+        sortable: false,
+        headerName: 'Group',
+        minWidth: 200,
+    };
+    const gridOptions: GridOptions<KxDocument> = {
+        columnDefs: colDefs,
+        rowGroupPanelShow: 'always',
+        animateRows: true,
+        pagination: false,
+        defaultColDef: {
+            filter: true,
+            flex: 1,
+            resizable: true,
+            sortable: true,
+            enableRowGroup: true,
+
+        },
+        sideBar: {
+            toolPanels: [
+                {
+                    id: 'columns',
+                    labelDefault: 'Columns',
+                    labelKey: 'columns',
+                    iconKey: 'columns',
+                    toolPanel: 'agColumnsToolPanel',
+                },
+                {
+                    id: 'filters',
+                    labelDefault: 'Filters',
+                    labelKey: 'filters',
+                    iconKey: 'filter',
+                    toolPanel: 'agFiltersToolPanel',
+                },
+            ],
+        },
+        autoGroupColumnDef: autoGroupColumnDef,
+    };
+
+    function onGridReady() {
+        gridRef.current!.api.sizeColumnsToFit();
+    }
+
+    const rowData = useMemo(() => {
+        return props.documents;
+    }, [props.documents]);
+
+   
+
+    const defaultColDef: ColDef = {
+        flex: 1,
+    };
+
+    return (
+        <div
+            className={
+                "ag-theme-quartz-auto-dark"
+            }
+            style={{ width: "100%", height: "100%" }}
+        >
+            <AgGridReact
+                rowData={rowData}
+                columnDefs={colDefs}
```

Comment:
there are about 8 errors for me on the console related the wrong use of hook. As the other's include the pictures
<img width="1437" alt="Screen Shot 2024-11-09 at 11 00 24" src="https://github.com/user-attachments/assets/8e76c403-966b-41b6-9ac7-a9d3f81d861a">

Reply:
That refers to a license issue, I've already asked for a edu license